### ISSUE TITLE
Multiple Third Party controllers Motion and Rumble stopped Working on SDL 2

### ISSUE THREAD

**Issue by ygordreyer** [libsdl-org/SDL#7577]
This is entirely based on this Issue that I've reported on CEMU : https://github.com/cemu-project/Cemu/issues/746

But in short : third Party controllers with multiple functionalities are not returning has_motion() anymore on version 2.26.4.
Previous version used by CEMU (2.0.20) was working fine.

Currently it's not working on:
8Bitdo Ultimate Bluetooth/2.4
8Bitdo Pro 2
Gulikit KingKong Pro

Working with original Joycons, and Switch Pro Controllers.

Since all this issue it's described in the CEMU Github, I won't copy and paste here and have a duplicated issue. 

If there is any additional info that I can provide that would help out, please let me know!

**Comment by slouken** [comment]
Did you mean to close this issue?

**Comment by Squall-Leonhart** [comment]
Yes, the user opened this prematurely before the full extent of the issue was investigated, and another tester confirmed their own gulikit to work.

**Comment by Squall-Leonhart** [comment]
Can this be reopened?, there is some sporadic reproduction of this with ryujinx and a Cemu user has reported that their affected control has a guid change between the last working version (on 2.0.18) and first version using 2.26.x

> After updating to version 2.0-30, the UUID is different from version 2.0-29, requiring the SDL controllers to be reconfigured. However, after reconfiguring, the vibration and vibration functions cannot be used. In version 2.0-29, the UUID was 0_030056fb7e0500000920000000006803, but in version 2.0-30, it changed to 0_030073747e0500000920000000000000. Is that the case for you?

**Comment by slouken** [comment]
This is caused by SDL not being able to open the raw devices anymore. It's likely that the user installed udev rules to block the raw devices to prevent Steam from opening them.

**Comment by Squall-Leonhart** [comment]
> This is caused by SDL not being able to open the raw devices anymore. It's likely that the user installed udev rules to block the raw devices to prevent Steam from opening them.

The issue is occuring on windows, there no udev rules to apply :p

**Comment by slouken** [comment]
I just tested the latest version of SDL (from main) on Windows 11, and discovered the following behavior:
* 8Bitdo Ultimate Bluetooth over USB: This controller enumerates over USB as an Xbox controller (and the home button brings up the Microsoft Game Bar) and the controller sends no gyro information to the computer.
* 8Bitdo Ultimate Bluetooth over Bluetooth: This controller shows up as a Nintendo Switch Pro controller and fails to open with SDL 2.0.20 (couldn't get device info), and opens as a DirectInput controller that doesn't provide gyro data with 2.27.0 (couldn't get device info for the HIDAPI driver and skipped enumeration there). If I hack around the controller initialization, the controller sends simple format reports that don't include gyro data.
* 8BitDo Pro 2 over USB: This controller shows up as a Nintendo Switch Pro controller with full gyro functionality
* 8BitDo Pro 2 over Bluetooth: This controller shows up as a Nintendo Switch Pro controller with full gyro functionality

**Comment by slouken** [comment]
After updating the firmware on my 8BitDo Ultimate Bluetooth controller from 1.02 to 1.03, it shows up over Bluetooth as a Nintendo Switch Pro controller with full gyro functionality.

**Comment by slouken** [comment]
I downloaded Cemu 29 and 30 and tested both of them with the 8Bitdo Ultimate Bluetooth controller with updated firmware, and I was able to select "Use motion" in both versions. I had the same result testing with the 8BitDo Pro 2.

So it sounds like we're getting different results here. You can download the SDL source from the [releases](https://github.com/libsdl-org/SDL/releases) and run the testgamecontroller test, to compare the results with what you're seeing in Cemu.

**Comment by Squall-Leonhart** [comment]
the ZD-O (Wired) is also failing to show support for Motion in NS(Nintendo Switch) mode.

**Comment by Squall-Leonhart** [comment]
> I just tested the latest version of SDL (from main) on Windows 11, and discovered the following behavior:

Could it be a flaw in Windows 10 presently?

**Comment by slouken** [comment]
There have been a bunch of fixes in the latest code, can you try with the current code on GitHub?

I will be creating a 2.28 prerelease next week if you’d like to try it out with that.

**Comment by slouken** [comment]
The SDL 2.28 prerelease is ready, can you please confirm whether or not this is fixed?
https://github.com/libsdl-org/SDL/releases/tag/prerelease-2.27.1

**Comment by Squall-Leonhart** [comment]
Haven't heard back from Ygor, but the Yuzu user reported it fixed, and several others have also reported motion drift to also be fixed.

**Comment by slouken** [comment]
Okay, thanks!

**Comment by german77** [comment]
Yuzu has build in motion drift correction so maybe the issue is not solved but masked. I just noticed I forgot to use user calibration instead of factory calibration for motion on https://github.com/libsdl-org/SDL/pull/7037. If this is going to be on 2.28.0  I can PR it in a few minutes.

**Comment by german77** [comment]
Never mind just looked at the code. It already has support for user calibration.
